### ISSUE TITLE
About Inference Speed Regarding Yolov5l

### ISSUE THREAD

**Issue by callbarian** [ultralytics/yolov5#5468]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi, I have question on an inference speed regarding batch 1 and 32.
My machine is Rtx3090 (cuda11.0, pytorch 1.7.1), and the image is on size 640.

Yolov5l.pt on batch 1 is about 13ms, which is comparable to your 10ms on v100.
However, on batch 32 is about 250ms, and thus about 7.8ms per image, which is not comparable to your 2.7ms

I expected the batch inference would give accelerated inference time, but not much improvement compared to batch 1.

detect.py --weights yolov5l.pt --source imagesource 

Is there anything I am missing?

Thank you.

**Comment by glenn-jocher** [comment]
@callbarian from v6.0 release PR https://github.com/ultralytics/yolov5/pull/5141:

## About Reported Speeds

mAP values are reproducible across any hardware, but **speeds will vary significantly among V100 instances**, and seem to depend heavily on the CUDA, CUDNN and PyTorch installations used. The numbers reported above were produced on GCP [N1-standard-8](https://cloud.google.com/compute/docs/machine-types) Skylake V100 instances running the [v6.0 Docker image](https://hub.docker.com/layers/ultralytics/yolov5/v6.0/images/sha256-58fe76178d82740e5e1c87b382590c7eca5beb952c8b7bab50b5cadeac6eb967?context=explore) with:

- NVIDIA Driver Version: 460.73.01
- CUDA Version: 11.2 

```bash
# Pull and Run v6.0 image
t=ultralytics/yolov5:v6.0 && sudo docker pull $t && sudo docker run -it --ipc=host --gpus all $t
```

<img width="1048" alt="Screen Shot 2021-10-12 at 10 57 31 AM" src="https://user-images.githubusercontent.com/26833433/137006376-a0d8eba0-fe50-4110-b953-69f7b3eeb08d.png">

Our speed command is:
```bash
# Reproduce YOLOv5s batch-1 speeds in table
python val.py --data coco.yaml --img 640 --task speed --batch 1
```
<img width="1214" alt="Screen Shot 2021-10-12 at 11 03 34 AM" src="https://user-images.githubusercontent.com/26833433/137006804-b21c7391-af9b-4867-a41d-60a89235af9b.png">

We tried several options, including AWS P3 instances, pulling more recent base versions (`FROM nvcr.io/nvidia/pytorch:21.09-py3`), different pytorch install methods, and many returned slower speeds than the values reported in our table. batch-32 speeds were found to vary less across options.

**Comment by yanisIk** [comment]
@callbarian Just by curiosity, what OS are you using ?

**Comment by callbarian** [comment]
@yanisIk I am using Ubuntu 18.04

@glenn-jocher Thank you for your reply. I have figured out the gap between yours and my experiment.
val.py uses the model.half().
I used val.py for measuring time and replicated close to your result.

Also, onnxruntime copies the values from cpu to gpu for input, and gpu to cpu for output.
This takes quite amount of time, whereas pytorch input is already on gpu.

pytorch model.half() is comparable to tensorRt float16 inference.
